### PR TITLE
chore: multi-build super chain image with several Debian releases

### PR DESCRIPTION
**Pull request by TheRealAmazonKendra**
The Debian 10 image uses Python 3.7 which was causing failures elsewhere due to required minimum python 3.8. Debian 10 is also ten months out of security support so upgrading it seemed like the right decision here. This change also updates Node to 16 due to Node 14 being EOL..

This is the same as another PR posted but upgrading to Debian 12 instead.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by TheRealAmazonKendra** [comment]
I was concerned that upgrading twice in one go wouldn't work so I did both and am letting them build/test here to see.

**Comment by RomainMuller** [comment]
The image tagging also needs to be updated...

**Review by mrgrain — APPROVED** [review]

**Comment by TheRealAmazonKendra** [comment]
@mergifyio update

File: .mergify/config.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ pull_request_rules:
       - '#changes-requested-reviews-by=0'
       - status-success=Validate PR Title
       # Docker image validation
-      - status-success~=^jsii/superchain
+      - status-success~=^Docker Images/jsii/superchain
```
Comment (RomainMuller) [inline .mergify/config.yml:36]:
```suggestion
      - status-success~=^Done$
```

**Pull request merged**